server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -290,15 +290,33 @@ MappedFieldType failIfFieldMappingNotFound(String name, MappedFieldType fieldMap
 
     private SearchLookup lookup = null;
 
+    /**
+     * Get the lookup to use during the search.
+     */
     public SearchLookup lookup() {
         if (this.lookup == null) {
             this.lookup = new SearchLookup(
                 getMapperService(),
-                (fieldType, searchLookup) -> indexFieldDataService.apply(fieldType, fullyQualifiedIndex.getName(), searchLookup));
+                (fieldType, searchLookup) -> indexFieldDataService.apply(fieldType, fullyQualifiedIndex.getName(), searchLookup)
+            );
         }
         return this.lookup;
     }
 
+    /**
+     * Build a lookup customized for the fetch phase. Use {@link #lookup()}
+     * in other phases.
```

Comment:
shall we open an issue to track finding an alternative for this?

Reply:
It looks like @romseygeek  is working on this now.